cmd/web-handlers.go

```diff
@@ -667,10 +667,13 @@ next:
 	for _, objectName := range args.Objects {
 		// If not a directory, remove the object.
 		if !HasSuffix(objectName, SlashSeparator) && objectName != "" {
-			// Check for permissions only in the case of
-			// non-anonymous login. For anonymous login, policy has already
-			// been checked.
-			govBypassPerms := ErrAccessDenied
+			// Get the retention/lock info to check if it is enabled or not
+			retentionInfo, _ := globalBucketObjectLockConfig.Get(args.BucketName)
+			bucketObjectLockEnabled := retentionInfo.Mode != ""
+			fmt.Println("bucketObjectLockEnabled =", bucketObjectLockEnabled)
+
+			// Check permissions for non-anonymous user
+			govBypassPerms := ErrNone
```

Comment:
Default should be ErrAccessDenied implicit deny rule @ebozduman this should be reset to ErrNone after policy validation.

Reply:
Sure. I'll change it. This also means I have to toggle the `if !globalPolicySys.IsAllowed` conditions.